lib/rubocop/cop/rails/timecop.rb

```diff
@@ -0,0 +1,188 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop disallows all usage of `Timecop`, in favour of
+      # `ActiveSupport::Testing::TimeHelpers`.
+      #
+      # ## Migration
+      # `Timecop.freeze` should be replaced with `freeze_time` when used
+      # without arguments. Where a `duration` has been passed to `freeze`, it
+      # should be replaced with `travel`. Likewise, where a `time` has been
+      # passed to `freeze`, it should be replaced with `travel_to`.
+      #
+      # `Timecop.return` should be replaced with `travel_back`, when used
+      # without a block. `travel_back` does not accept a block, so where
+      # `return` is used with a block, it should be replaced by explicitly
+      # calling `freeze_time` with a block, and passing the `time` to
+      # temporarily return to.
+      #
+      # `Timecop.scale` should be replaced by explicitly calling `travel` or
+      # `travel_to` with the expected `durations` or `times`, respectively,
+      # rather than relying on allowing time to continue to flow.
```

Comment:
If the amount of time being measured and scaled up is large enough that Ruby can measure it at all, then it can be explicitly checked against, even if something like `nsec` is required.

In this case, I'm not sure that test would be checking anything worthwhile. It's coupling to how long the overhead of the stub is, rather than being explicit about what it's meant to be simulating.

Rather than `scale` time, I'd `travel`:

```ruby
Hubspot.stub(:track) do
  travel(0.5.seconds) # after_teardown ensures this is safe
end

started_at = Time.zone.now
Hubspot.track(...)
expect(Time.zone.now).to be_within(1.second).from(started_at)
```

This way, if a test is asserting that a quick API call works fine, and a slow one is handled however it must be handled, one can just make it appear to take the time it should.